### PR TITLE
[Doc Links Service] Clean up noDocumentation links

### LINKED ISSUE #3208: [Tech Debt] Clean up `docs_link_service` organization so that strings are in the right categories
**Is your feature request related to a problem? Please describe.**

In https://github.com/opensearch-project/OpenSearch-Dashboards/blob/68d503d110862d55baaf90a190ab17990e9e0f08/src/core/public/doc_links/doc_links_service.ts#L54

We organize links to documentation in 3 categories:

1. `opensearch` - these should be documentation links at https://opensearch.org/docs/latest/opensearch
2. `opensearchDashboards` - these should be documentation links at https://opensearch.org/docs/latest/dashboards
3. `noDocumentation` - these should only be documentation link placeholders where no valid documentation exists

**Describe the solution you'd like**

As far as I can tell, existing links in `opensearch` and `opensearchDashboards` are all correctly categorized and formatted.

But for links defined under `noDocumentation`:

1. If an actual documentation link is defined (more than just the documentation homepage)
  1. Replace any hard-coded path with `${OPENSEARCH_DASHBOARDS_VERSIONED_DOCS}` or `${OPENSEARCH_VERSIONED_DOCS}` as appropriate
  2. If there's no comment with the absolute URL to the doc link, add a comment above the property
  3. Move the property to `opensearch` or `opensearchDashboards` property as appropriate
  4. Update the `DocLinksStart` interface
  5. Update any references in code from `.noDocumentation.` to the appropriate property 
2. Some links that appear to be valid are not. For example https://opensearch.org/docs/latest/downloads/beats/auditbeat redirects to the documentation homepage. These links should be corrected if actual documentation pages exist, or updated to be `${OPENSEARCH_WEBSITE_DOCS}`, which will make it more clear that the documentation is really missing

### PR DESCRIPTION
**Pull request by sayuree**
### Description
[Added corresponding links from the documentation for non-working links
 Edited DocLinksStart interface properties
 Edited the usage of interface in other files correspondigly
]
 
### Issues Resolved
fixes #3208
 
### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by Abyl10** [comment]
Thank you for your recent code changes! I noticed that the ```CHANGELOG.md``` file was not updated to reflect the recent changes. It's important that we keep the changelog up-to-date so that other developers can understand what changes were made, why they were made, and how they may impact the codebase. Could you please update the ```CHANGELOG.md``` file with as much detail as possible?

**Review by joshuarrrr — CHANGES_REQUESTED** [review]
@sayuree Thanks for the PR - it mostly looks great!

I found some additional links that can be defined.

There are also some minor requested changes.

File: CHANGELOG.md [hunk 1]
```diff
@@ -230,6 +230,8 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 
 ### 🪛 Refactoring
 
+- [Tech Debt] Clean up docs_link_service organization so that strings are in the right categories. ([#3208](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3685))
```
[the remark this answers is not in the document]
Comment (sayuree) [reply CHANGELOG.md:233]: Yes, I have updated the link to PR in a later commit

**Review by joshuarrrr — COMMENTED** [review]
@sayuree Can you also add the commented links to the `restAPI` section (L330-340)?

File: src/plugins/data_source_management/public/management_app/compoenent/experimental_callout/experimental_callout.tsx [hunk 2]
```diff
@@ -40,7 +40,7 @@ export const ExperimentalCallout = ({ docLinks }: { docLinks: DocLinksStart }) =
               defaultMessage="To leave feedback, visit "
             />
           }
-          <EuiLink href={docLinks.links.noDocumentation.openSearchForum} target="_blank">
+          <EuiLink href={docLinks.links.opensearch.openSearchForum} target="_blank">
```
Comment (joshuarrrr) [inline src/plugins/data_source_management/public/management_app/compoenent/experimental_callout/experimental_callout.tsx:43]: Please also update the mock in src/plugins/data_source_management/public/mocks.ts

File: src/core/public/doc_links/doc_links_service.ts [hunk 3]
```diff
@@ -405,25 +408,24 @@ export class DocLinksService {
...
-          filebeat: `https://opensearch.org/docs/latest/downloads/beats/filebeat`,
-          metricbeat: `https://opensearch.org/docs/latest/downloads/beats/metricbeat`,
-          heartbeat: `https://opensearch.org/docs/latest/downloads/beats/heartbeat`,
+          auditbeat: `${OPENSEARCH_WEBSITE_DOCS}`,
+          filebeat: `${OPENSEARCH_WEBSITE_DOCS}`,
+          metricbeat: `${OPENSEARCH_WEBSITE_DOCS}`,
+          heartbeat: `${OPENSEARCH_WEBSITE_DOCS}`,
           logstash: `${OPENSEARCH_WEBSITE_DOCS}`,
-          functionbeat: `https://opensearch.org/docs/latest/downloads/beats/functionbeat`,
+          functionbeat: `${OPENSEARCH_WEBSITE_DOCS}`,
```
Comment (joshuarrrr) [inline src/core/public/doc_links/doc_links_service.ts:422]: For each of these, let's use `${OPENSEARCH_WEBSITE_DOCS}tools/index/#downloads`

**Comment by joshuarrrr** [comment]
@sayuree Will you be able to make the additional requested changes?

**Comment by sayuree** [comment]
@joshuarrrr Added requested changes

File: src/core/public/doc_links/doc_links_service.ts [hunk 4]
```diff
@@ -405,25 +408,28 @@ export class DocLinksService {
           },
           // https://opensearch.org/docs/latest/dashboards/browser-compatibility
           browser: `${OPENSEARCH_DASHBOARDS_VERSIONED_DOCS}browser-compatibility`,
+          dataSource: {
+            // https://opensearch.org/docs/latest/dashboards/discover/multi-data-sources/
+            guide: `${OPENSEARCH_DASHBOARDS_VERSIONED_DOCS}discover/multi-data-sources/`,
+          },
+          visualize: {
+            guide: `${OPENSEARCH_WEBSITE_DOCS}`,
```
Comment (joshuarrrr) [inline src/core/public/doc_links/doc_links_service.ts:416]:
```suggestion
            // https://opensearch.org/docs/latest/dashboards/visualize/viz-index/
            guide: `${OPENSEARCH_WEBSITE_DOCS}visualize/viz-index/`,
```

**Review by joshuarrrr — COMMENTED** [review]
Thanks @sayuree , almost there! Just three tasks left, summarized here:

1. Can you also add the commented links to the restAPI section (L345-355)?
2. Please also update the mock in src/plugins/data_source_management/public/mocks.ts
3. See suggested changes for `visualize.guide` link.

File: src/plugins/data_source_management/public/mocks.ts [hunk 5]
```diff
@@ -12,7 +12,7 @@ import { coreMock } from '../../../core/public/mocks';
 
 export const docLinks = {
   links: {
-    noDocumentation: {
+    opensearch: {
```
Comment (joshuarrrr) [inline src/plugins/data_source_management/public/mocks.ts:15]:
```suggestion
    noDocumentation: {
```

**Comment by joshuarrrr** [comment]
@sayuree Can you rebase or update the PR to resolve the conflict? I believe the experimental callout has since been removed.

**Comment by sayuree** [comment]
> @sayuree Can you rebase or update the PR to resolve the conflict? I believe the experimental callout has since been removed.

I have updated the PR, hope changes will be approved.

**Review by joshuarrrr — APPROVED** [review]

**Review by AMoo-Miki — APPROVED** [review]

**Pull request merged**